### PR TITLE
 Add and Remove Task

### PR DESCRIPTION
**Pull request by DamenaM**
### PR for  Add and Remove Task
- [ ] Implement a function for adding a new task (add a new element to the array).
- [ ] Implement a function for deleting a task (remove an element from the array).
- [ ] Implement a function for editing task descriptions.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,41 +1,113 @@
 // import _ from 'lodash';
```
Comment (huseyinbiyik) [inline src/index.js:1]: Kindly add a new module to your repository for the new functionalities as requested on project requirements.
Screenshot 2023-02-01 at 11 23 21 PM

File: src/index.html [hunk 2]
```diff
@@ -1,36 +1,58 @@
...
-       </div>
-
-    </div>
+          <div class="container-form">
+                
+             <div class="wrapper">
+              <form action="" id="new-task" class="task-container row">
+                <input type="text" class="input" placeholder="Add to your list..." id="task-description">
+                <!-- <input type="text" name="title" id="title" placeholder="Title" class="input_title"> <br>
+                <input type="text" name="Author" placeholder="Author" id="author" class="input_author"> <br> -->
```
Comment (huseyinbiyik) [inline src/index.html]: Please don't leave any uncommented code in your codebase. In this way, you can keep your codebase neat, clean, and professional. 👨‍💼

File: src/index.js [hunk 3]
```diff
@@ -1,41 +1,113 @@
...
+    this.checkboxStatus = document.querySelector('#checkboxid');
+    this.render();
+  }
+
+  addTask(id, title, status) {
+    this.tasks.push({ id, title, status });
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+    this.render();
+  }
+
```
Comment (huseyinbiyik) [inline src/index.js]: Good job with adding and deleting tasks. But kindly add edit task functionality as requested on project requirements.
Screenshot 2023-02-01 at 11 31 13 PM

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi @DamenaM,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
Almost there

## Highlights
- Add and delete functionality works 👌
- Linter set up well and checks pass 🟢
- Github flow followed 👌
- PR description is descriptive 📝

## Required Changes ♻️
-  Please make sure your commit messages are meaningful and descriptive for your next commits. Use imperative tense and capitalize the first word. For example, 
👌"Add logo to navigation bar"
👌"Fix close button functionality on mobile navbar"
❌ "Added logo to navigation bar"
❌ "Added HTML"
_Check the comments under the review._

##OPTIONAL
- I think it would better if you can remove and update some of texts in your README file. For example, you don't need to thank you Cindy on each project. But if you want to thank her, no problem  👍
Screenshot 2023-02-01 at 11 39 06 PM
Screenshot 2023-02-01 at 11 38 56 PM



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Module/taskapp.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      this.render();
+    }
+
+    removeTask(index) {
+      this.tasks.splice(index, 1);
+      localStorage.setItem('tasks', JSON.stringify(this.tasks));
+      this.render();
+    }
+
+    editTask(target) {
```
Comment (huseyinbiyik) [inline src/Module/taskapp.js]: Good job with adding the editing task. But let's implement the edit behaviour as seen in the video as requested [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/sneak_peek.md#how-to-build-the-to-do-list-website). 
>Your task is to build a website that displays a list that looks and behaves like the part of minimalist project capture in the video below.

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi @DamenaM 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
tenor

 ## Highlights
- Linter set up well and checks pass 🟢
- Github flow followed 👌


 ## Required Changes ♻️
-  Please make sure your commit messages are meaningful and descriptive for your next commits. Please use imperative tense and capitalize the first word. Kindly check the [Professional Repository](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) document.  For example, 
👌"Add logo to navigation bar"
👌"Fix close button functionality on mobile navbar"
❌ "Added logo to navigation bar"
❌ "Added HTML"

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Module/taskapp.js (hunk 4 above)
Comment (Favourezeugwa) [reply src/Module/taskapp.js]: - I think your function for editing input has not been correctly implemented. just as @huseyinbiyik suggested, can you fix this?? 🌸 
When I try to edit, and click enter, the terms "edit" and "delete"  automatically add to the newly edited input. Here is what I mean 👇🏽 

Screenshot 2023-02-03 at 23 02 29

**Review by Favourezeugwa — CHANGES_REQUESTED** [review]
## STATUS: REQUIRES CHANGES 🚦 ⚠️ ♻️ 

Hi @DamenaM  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


close lady

## Highlights 💯 
- No Linter errors ✔️
- Descriptive Pull request summary✔️
- Good commit messages ✔️
- Professional Readme file ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @DamenaM  
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- All linters checks are ok. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**